SOURCES/etc/xapi.d/plugins/updater.py

```diff
@@ -1,16 +1,84 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
+from functools import wraps
+import ConfigParser
+import json
+import os
 import subprocess
 import sys
 import traceback
 import XenAPIPlugin
 import yum
-import ConfigParser
-import json
 
 sys.path.append('.')
-from xcpngutils import configure_logging
+from xcpngutils import configure_logging, run_command
+from xcpngutils.filelocker import FileLocker
+
+
+REPOS = ('xcp-ng-base', 'xcp-ng-updates')
+
+
+class OperationException(Exception):
+    pass
+
+class OperationLocker(FileLocker):
+    __slots__ = ('operation', 'current_operation')
+
+    def __init__(self, operation, timeout=0):
+        super(OperationLocker, self).__init__(
+            timeout=timeout,
+            auto_remove=False,
+            dir=os.path.dirname(os.path.abspath(__file__))
+        )
+        self.operation = operation
+
+    def _raise_busy(self):
+        raise OperationException('The updater plugin is busy (current operation: {})'.format(self.current_operation))
+
+    def _prelock(self):
+        self.current_operation = None
+        try:
+            self.file.seek(0)
+            self.current_operation = self.file.readline().rstrip('\n')
+        except:
+            pass
+
+        if not self.current_operation:
+            return
+
+        selfIsUpdate = self.operation == 'update'
```

Comment:
`self_is_update` might be more consistent with other variable names

Reply:
The result of human thoughtlessness. :thinking: